### PR TITLE
Game options menu

### PR DESCRIPTION
**Pull request by RabeaGleissner**
@felipesere @jsuchy

Here's my PR for the Artisan story about the game options menu. 
The user can now choose from the following game options:
- HvH
- HvC
- CvH
- CvC

Let me know what you think! Thanks.

### CODE REVIEW

File: src/main/java/de/rabea/game/GameMode.java [hunk 1]
```diff
@@ -2,5 +2,7 @@
 
 public enum GameMode {
     HvH,
-    HvC
+    HvC,
+    CvH,
+    CvC
```
Comment (octopusinvitro) [inline src/main/java/de/rabea/game/GameMode.java]: I would use a bit more verbose names maybe. I think it would be more readable.

**Comment by octopusinvitro** [comment]
LGTM :+1:

**Comment by RabeaGleissner** [comment]
@me-stevens Thanks for the feedback! I updated the names of the Enums.

**Comment by RabeaGleissner** [comment]
@felipesere
I'll merge this PR into master (as agreed with Jim), so that I can use the code for when I start working on my next story. I won't delete it so you should still be able to comment.

File: src/test/java/de/rabea/game/GameTypeTest.java [hunk 2]
```diff
@@ -12,29 +14,83 @@
...
     }
 
     @Test
-    public void itSetsUpTheFirstGameAsAHvHGame() {
-        fakeUserInterface.provideConsoleInput("2");
-        Game game = gameType.createGame();
-        assertTrue(game != null);
+    public void createsAHumanVsHumanGame() {
+        GameType gameType = new GameType(new UIReturningHvH(fakeConsole), gameManager);
+        assertTrue(gameType.createGame() != null);
```
Comment (jsuchy) [inline src/test/java/de/rabea/game/GameTypeTest.java:31]: I don't like tests that check x != null. Can you actually test some behaviour in these tests?

Comment (RabeaGleissner) [inline src/test/java/de/rabea/game/GameTypeTest.java:31]: @jsuchy I can't think of a good way to test this. Could you give me a hint? 

The only two things I can think of are to 
- create a "getPlayer()" method in Game and then assert for the type of player that is created
- make the createPlayer() and createOpponent() methods public and then test for the type of player that is created

But I don't think either options are very good...

**Pull request merged**